src/platform_impl/linux/x11/window.rs

```diff
@@ -649,10 +659,12 @@ impl UnownedWindow {
                 let (video_mode, monitor) = match fullscreen {
                     Fullscreen::Exclusive(RootVideoMode {
                         video_mode: PlatformVideoMode::X(ref video_mode),
-                    }) => (Some(video_mode), video_mode.monitor.as_ref().unwrap()),
-                    Fullscreen::Borderless(RootMonitorHandle {
-                        inner: PlatformMonitorHandle::X(ref monitor),
-                    }) => (None, monitor),
+                    }) => (Some(video_mode), video_mode.monitor.clone().unwrap()),
+                    Fullscreen::Borderless(Some(RootMonitorHandle {
+                        inner: PlatformMonitorHandle::X(monitor),
+                    })) => (None, monitor),
+                    // We've mapped `None` in borderless earlier.
+                    Fullscreen::Borderless(None) => unreachable!(),
```

Comment:
This looks a bit unfortunate to me. Why does it need to be remapped earlier instead of handled here? So comparison to the existing state works? I don't really see where the monitor is needed in the old/new comparison.